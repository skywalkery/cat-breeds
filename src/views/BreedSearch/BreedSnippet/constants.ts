/* eslint-disable import/prefer-default-export */
export const DEFAULT_IMG_URL = '/static/icons/cat.svg';
