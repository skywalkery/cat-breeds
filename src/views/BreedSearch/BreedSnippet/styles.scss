.container {
  width: 100%;
  max-width: 50rem;
  position: relative;
  border-radius: 1.6rem;
  overflow: hidden;
  transition: box-shadow 0.3s;

  @media (min-width: 48em) {
    width: 35rem;
  }

  @media (min-width: 66.75em) {
    width: 32.4rem;
  }

  &:hover {
    box-shadow: 0 1px 11px -4px rgba(0, 0, 0, 0.2);

    .imgWrapper {
      > img {
        transform: scale(1.1);
      }
    }
  }
}

.imgWrapper {
  position: relative;
  overflow: hidden;

  > img {
    width: 100%;
    height: 31rem;
    object-fit: cover;
    transform: scale(1);
    transition: transform 0.3s;
  }
}

.titles {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
  padding: 0.5rem;
  background: rgba(255, 255, 255, 0.6);
  backdrop-filter: blur(5px);
}

.name {
  font-size: 2.4rem;
  line-height: 3.2rem;
}

.description {
  padding: 0.5rem;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 5;
}
