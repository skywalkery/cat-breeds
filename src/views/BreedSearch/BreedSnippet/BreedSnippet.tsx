import { FC, memo } from 'react';

import type { Breed } from '@ducks/breeds/types';
import { DEFAULT_IMG_URL } from './constants';
import styles from './styles.scss';
import BreedRating from './BreedRating';

type Props = {
  breed: Breed;
};

const BreedSnippet: FC<Props> = ({
  breed: { energyLevel, intelligence, name, description, imageUrl },
}) => {
  return (
    <div className={styles.container}>
      <div className={styles.imgWrapper}>
        <img src={imageUrl || DEFAULT_IMG_URL} alt="Cat img" />
        <div className={styles.titles}>
          <span className={styles.name}>{name}</span>
          <div>
            <span>Energy Level: </span>
            <BreedRating value={energyLevel} />
          </div>
          <div>
            <span>Intelligence: </span>
            <BreedRating value={intelligence} />
          </div>
        </div>
      </div>
      <div className={styles.description}>{description}</div>
    </div>
  );
};

export default memo(BreedSnippet);
