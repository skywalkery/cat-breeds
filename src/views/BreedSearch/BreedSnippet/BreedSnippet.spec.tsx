/* eslint-disable import/no-extraneous-dependencies */
import renderer from 'react-test-renderer';

import type { Breed } from '@ducks/breeds/types';
import BreedSnippet from './BreedSnippet';

const breed: Breed = {
  energyLevel: 5,
  id: '1',
  intelligence: 4,
  lowerName: 'aaa',
  name: 'Aaa',
  description: 'Lorem iprum dolorem',
  imageUrl: '/img.jpg',
};

describe('Breed Snippet', () => {
  it('renders properly', () => {
    const component = renderer.create(<BreedSnippet breed={breed} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
