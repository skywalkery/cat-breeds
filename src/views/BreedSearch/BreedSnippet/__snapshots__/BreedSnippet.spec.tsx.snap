// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breed Snippet renders properly 1`] = `
<div
  className="container"
>
  <div
    className="imgWrapper"
  >
    <img
      alt="Cat img"
      src="/img.jpg"
    />
    <div
      className="titles"
    >
      <span
        className="name"
      >
        Aaa
      </span>
      <div>
        <span>
          Energy Level: 
        </span>
        <span
          style={
            Object {
              "direction": "ltr",
              "display": "inline-block",
            }
          }
        >
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
        </span>
      </div>
      <div>
        <span>
          Intelligence: 
        </span>
        <span
          style={
            Object {
              "direction": "ltr",
              "display": "inline-block",
            }
          }
        >
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={
                Object {
                  "visibility": "hidden",
                }
              }
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
          <span
            onClick={[Function]}
            onMouseMove={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            style={
              Object {
                "cursor": "inherit",
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <span
              style={Object {}}
            >
              <img
                alt="Empty icon"
                className="icon"
                src="/static/icons/paw-empty.png"
              />
            </span>
            <span
              style={
                Object {
                  "display": "inline-block",
                  "left": 0,
                  "overflow": "hidden",
                  "position": "absolute",
                  "top": 0,
                  "width": "0%",
                }
              }
            >
              <img
                alt="Fill icon"
                className="icon"
                src="/static/icons/paw-fill.png"
              />
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="description"
  >
    Lorem iprum dolorem
  </div>
</div>
`;
